cmd/convox/uninstall_test.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/convox/rack/test"
+)
+
+var uninstallUsage = `convox uninstall: uninstall a convox rack
+
+Usage:
+  convox uninstall <stack-name> <region> [credentials.csv]
+
+Options:
+   --force  uninstall without verification prompt
+   
+`
+
+var errUninstallNeedArg = `ERROR: at least 2 arguments are required: <stack-name> <region>
+`
+var errNoSuchHost = `ERROR: RequestError: send request failed
+caused by: Post https://cloudformation.invalidRegion.amazonaws.com/: dial tcp: lookup cloudformation.invalidRegion.amazonaws.com on 192.168.1.1:53: no such host
+`
+
+func TestUninstallHelpFlag(t *testing.T) {
+	ts := testServer(t,
+		test.Http{
+			Method:   "GET",
+			Path:     "/foo",
+			Code:     200,
+			Response: "bar",
+			Headers:  map[string]string{"Rack": "myorg/staging"},
+		},
+	)
+	defer ts.Close()
+
+	tests := []test.ExecRun{
+		// help flags
+		test.ExecRun{
+			Command: "convox uninstall -h",
+			Stdout:  uninstallUsage,
+		},
+
+		// invalid usage
+		test.ExecRun{
+			Command: "convox uninstall",
+			Stdout:  uninstallUsage,
+			Exit:    129,
+		},
+		test.ExecRun{
+			Command: "convox uninstall onlyOneArgument",
+			Stdout:  uninstallUsage,
+			Exit:    129,
+		},
+
+		// FIXME: this command actually raises a RequestError
+		test.ExecRun{
+			Command: "convox uninstall foo invalidRegion",
+			Exit:    0,
+			Stdout:  Banner + "\n",
+			// Stderr: errNoSuchHost,
+		},
+
+		// FIXME: this command actually exits with a 'no such file or directory' error
+		test.ExecRun{
+			Command: "convox uninstall rackArg regionArg credentialsArg",
+			Stdout:  Banner + "\nReading credentials from file credentialsArg\n",
+			//Stderr: `ERROR: open credentialsArg: no such file or directory`,
```

Comment:
As I mentioned in the comment directly above `test.ExecRun`, when I run this command for real, the CLI exits 1 with `no such file or directory`. The test, however, doesn't capture that in either Stderr or Stdout, and I'm not sure why. Removing the commented `Stderr` anyway though.